### ISSUE TITLE
Training with smaller batch sizes

### ISSUE THREAD

**Issue by amirbar** [antoine77340/MIL-NCE_HowTo100M#9]
Hi @antoine77340 ,

Thank you very much for publishing this great paper and code. 

I'm wondering if you can share some insights regarding the effect of batch size on the results. Did you ever attempt to go below batch size 512? (e.g., 32/64/128?). While trying to reproduce the results, I wasn't able to get very far with batch size 64 (unfortunately, larger batch sizes are out of reach for me w.r.t compute).

Thanks!!
Amir

**Comment by antoine77340** [comment]
Hi!
Thank you for your comment.
I am not sure if we tried getting below batch size of 64.
But I can assume that the learning can be very difficult in that regime due to high noise in narration whereas batch size of 64 is usually fine with strong and clean supervision (like Kinetics).
Did you try putting a lower learning rate? I guess that would be important given that gradients would be very noisy with batch size of 64.

**Comment by amirbar** [comment]
Thank you for the fast response! I've trained with 1e-3/1e-4 for around 4 days with batch size 64. I've tested the model on HMDB and I get an accuracy of around 20%. I agree that the gradients might be noisier (or have larger variance more precisely). However, the denominator in the MIL-NCE loss actually relies on the size of the batch to get more negative samples, so this is not just noisy gradients, the expectation over the loss with different batch sizes should be different. I'm wondering if that's the reason.

**Comment by antoine77340** [comment]
That's right, this also have an effect on the number of negative examples which has an importance.
We do have an ablation study in the original paper where we vary the number of negatives an we tried with 64.
The results are still quite good on HMDB (56% in accuracy).

**Comment by amirbar** [comment]
Thank you for the helpful comments @antoine77340 !

**Comment by amirbar** [comment]
> the number of negatives an we tried with 64
@antoine77340 , in this setup, did you fine tune the representations or used fixed representations and just trained a linear classifier?

**Comment by antoine77340** [comment]
Hi!
i am sorry for this very late reply, this totally slipped by!
The representation was still learnt from scratch.
